src/object.cpp

```diff
@@ -393,6 +417,15 @@ int64_t ddwaf_object_get_signed(const ddwaf_object *object)
     return object->intValue;
 }
 
+double ddwaf_object_get_float(const ddwaf_object *object)
+{
+    if (object == nullptr || object->type != DDWAF_OBJ_FLOAT) {
+        return 0;
```

Comment:
How do user of this function know if the 0 comes from the object or from this conditional?

Reply:
They can't know, although they can always check if their object is null or of an incorrect type, but it's mostly a precaution for incorrect use of the API.